### ISSUE TITLE
Bug with nonempty_list() and nonempty_string()

### ISSUE THREAD

**Issue by skogsbaer** [etylizer/etylizer#183]
```erlang
-module(nonempty_list).

-compile(export_all).
-compile(nowarn_export_all).

-spec id(nonempty_list(integer())) -> nonempty_list(integer()).
id(S) -> S.

-spec foo() -> list(integer()).
foo() -> id([1,2,3]).
```

wrongly fails with

```
test_files/tycheck/nonempty_list.erl:10:13: Type error: expression failed to type check
%   10| foo() -> id([1,2,3]).
%     |
```

Similarly, 

```erlang
-module(nonempty_string).

-compile(export_all).
-compile(nowarn_export_all).

-spec id(nonempty_string()) -> nonempty_string().
id(S) -> S.

-spec foo() -> string().
foo() -> id("blub").
```

wrongly fails with

```
test_files/tycheck/nonempty_string.erl:10:13: Type error: expression failed to type check
%   10| foo() -> id("blub").
%     |
```

**Comment by albsch** [comment]
The constraints generated are:

```erlang
                        2 < $1
                     [$2] < nonempty_list(integer())
                     [$1] < nonempty_list(integer())
 nonempty_list(integer()) < [integer()]
                       [] < nonempty_list(integer())
                        3 < $2
                        1 < $0
                     [$0] < nonempty_list(integer())
```

Tally fails because the empty list is not included in a non-empty list. 

```erlang
 [] < nonempty_list(integer())
```

All other list constraints evaluate to false, too. The constraints look wrong, can you check @skogsbaer ? Likely related to #9.

The `erlang_types` library has the following assumption:

```
[X] == list(X) == nonempty_list(X) U '[]' == improper_list(X, '[]') == nonempty_improper_list(X, '[]') U '[]'
```

**Comment by skogsbaer** [comment]
fixed in #184
